2015/proposals/girdhar-yashasvi.md

```diff
@@ -0,0 +1,254 @@
+#Result-aggregation server for the installation-test scripts
+
+##Personal Details:
+
+* Name: Yashasvi Girdhar
+* Email: yash.girdhar@gmail.com
+* Personal Webpage : http://about.me/yashgirdhar
+* Github profile : https://github.com/itsyash
+* IRC nick (on freenode) : yash_
+* Telephone: +91-8143914414
+* Country of residence: India
+* Timezone: IST (GMT+5:30)
+* Primary language: English
+
+##Abstract:
+
+Software Carpentry has installation-test scripts which are used by students to check if they have correctly installed the required software for a workshop.
+
+The script basically performs a test for each software and outputs the result. In case of a failure, it prints the cause of failure and some diagnostic information on how to install the software correctly.
+Until now, we do not collect these results anywhere. But there are number of reasons why we should be doing that.
+
+Some sample queries that the system would serve will look like:
+
+* How many tests failed for a given workshop for students running Windows.
+* What fraction of failed tests were on OS X for a specific time period.
+* For a given workshop, find number of failed version tests for Git.
+
+Statistics like these will help us figure out more information about:
+* How much trouble the students are having with the installation.
+* Which dependencies need more of our attention (deprecating their older versions or adapting with the new versions).
+
+##Technical Details:
+
+The project is mainly divided into four phases:
+
+#### 1) A model for storing the data in a relational database.
+
+We start with deciding what information is useful and how should we store it. I have already built a first draft of the [schema](#schema) (added at the end of this proposal) with feedback from the mentor.
+I would use SQLite database for storing the data.
```

Comment:
Yes, at first,I would be using SQLite for test. I have kept a time period in my proposal to thoroughly test the API (before I start the front end) and I think we can switch the database at that time if we feel the need to do so.